tests/Unit/NumericalAlgorithms/Spectral/Test_SwshCollocation.cpp

```diff
@@ -0,0 +1,149 @@
+  CHECK(precomputed_collocation.end() == precomputed_collocation.cend());
+  CHECK(precomputed_collocation.begin() != precomputed_collocation.end());
+  for (const auto& collocation_point : precomputed_collocation) {
+    CHECK(collocation_point.offset == offset_counter);
+    CHECK(collocation_point.theta ==
+          computed_collocation.theta(offset_counter));
+    CHECK(collocation_point.phi == computed_collocation.phi(offset_counter));
+    CAPTURE(offset_counter);
+    // check theta values:
+    if (offset_counter < (2 * l_max + 1) * (l_max / 2 + 1)) {
+      CAPTURE(offset_counter / (2 * l_max + 1));
+      CHECK(approx(computed_collocation.theta(offset_counter)) ==
+            computed_sgi  // NOLINT
+                ->pair[offset_counter / (2 * l_max + 1)]
+                .r1.theta);  // NOLINT
+      CHECK(approx(computed_collocation.theta(offset_counter)) ==
+            manual_sgi  // NOLINT
+                ->pair[offset_counter / (2 * l_max + 1)]
+                .r1.theta);  // NOLINT
+    } else {
+      CAPTURE(l_max - (offset_counter / (2 * l_max + 1)));
+      CHECK(approx(computed_collocation.theta(offset_counter)) ==
+            computed_sgi  // NOLINT
+                ->pair[l_max - offset_counter / (2 * l_max + 1)]
+                .r2.theta);  // NOLINT
+      CHECK(approx(computed_collocation.theta(offset_counter)) ==
+            manual_sgi  // NOLINT
+                ->pair[l_max - offset_counter / (2 * l_max + 1)]
+                .r2.theta);  // NOLINT
+    }
+
+    // check phi values:
+    CHECK(approx(computed_collocation.phi(offset_counter)) ==
+          2.0 * M_PI *
+              ((offset_counter % (2 * l_max + 1)) / (2.0 * l_max + 1.0)));
+    CHECK(approx(computed_collocation.phi(offset_counter)) ==
+          2.0 * M_PI *
+              ((offset_counter % (2 * l_max + 1)) / (2.0 * l_max + 1.0)));
+    offset_counter++;
+  }
+}
+
+SPECTRE_TEST_CASE("Unit.NumericalAlgorithms.Spectral.SwshCollocation",
+                  "[Unit][NumericalAlgorithms]") {
+  {
+    INFO("Collocation based on contiguous complex data (Interleaved)");
+    test_spherical_harmonic_collocation<ComplexRepresentation::Interleaved>();
+  }
+  {
+    INFO(
+        "Collocation based on a pair of contiguous blocks representing complex "
+        "data (RealsThenImags)");
+    test_spherical_harmonic_collocation<
+        ComplexRepresentation::RealsThenImags>();
+  }
+}
+
+// [[OutputRegex, is not below the maximum l_max]]
+[[noreturn]] SPECTRE_TEST_CASE(
+    "Unit.NumericalAlgorithms.Spectral.SwshCollocation.PrecomputationOverrun",
+    "[Unit][NumericalAlgorithms]") {
+  ERROR_TEST();
+  const Collocation<ComplexRepresentation::RealsThenImags>&
+      precomputed_collocation = precomputed_spherical_harmonic_collocation<
+          ComplexRepresentation::RealsThenImags>(collocation_maximum_l_max + 1);
+  ERROR("Failed to trigger ASSERT in an assertion test");
```

Comment:
Woops, this `ERROR` can go :)

Reply:
I think the compiler needs an error here to avoid generating warnings that the noreturn might return. I've updated the error to appropriately refer to `ERROR in an error test`.